ruby/automated_testing/test_driven_development.md

```diff
@@ -218,7 +221,7 @@ Read ["TDD is Dead. Long Live Testing"](http://david.heinemeierhansson.com/2014/
 
 This section contains questions for you to check your understanding of this lesson. If you’re having trouble answering the questions below on your own, review the material above to find the answer.
 
-- What does it mean for code to be test **driven**?
-- List four different advantages of TDD.
-- What is the 3 part development cycle used for TDD?
-- How much code should be written when going from the `red` state to the `green` state of the red-green-refactor cycle?
+- [What does it mean for code to be test **driven**?](#what-is-tdd-and-why-do-it)
+- [List four different advantages of TDD](#what-is-tdd-and-why-do-it)
+- [What is the 3 part development cycle used for TDD?](#what-is-tdd-and-why-do-it)
+- [How much code should be written when going from the `red` state to the `green` state of the red-green-refactor cycle?](#what-is-tdd-and-why-do-it)
```

Comment:
Having link backs in the Knowledge Check section is good, but if they all point to the same section, then they might cause confusion when they all end up at the same place.

Additionally, you've missed a period at the end of the second check.